include/options.hxx

```diff
@@ -191,21 +370,57 @@ public:
 
   /// Read-only access to internal options and sections
   /// to allow iteration over the tree
-  const std::map<string, OptionValue>& values() const {return options;}
-  const std::map<string, Options*>& subsections() const {return sections;}
+  std::map<string, OptionValue> values() const;
+  std::map<string, Options*> subsections() const;
 
  private:
-  static Options *root; ///< Only instance of the root section
+  
+  static Options *root_instance; ///< Only instance of the root section
 
-  Options *parent;
-  string sectionName; // section name (if any), for logging only
+  Options *parent_instance;
+  string full_name; // full path name for logging only
 
-  std::map<string, OptionValue> options;
-  std::map<string, Options*> sections;
+  /// An Option object can be a section and/or a value, or neither (empty)
 
-  void _set(const string &key, const string &val, const string &source, bool force);
+  bool is_section = false; ///< Is this Options object a section?
+  std::map<string, Options*> children; ///< If a section then has children
+
+  bool is_value = false; ///< Is this Options object a value?
+  OptionValue value; ///< If a value
```

Comment:
Does it make sense to give this a default empty value (`OptionValue value{}`)? We would need to give `OptionValue::used` a default value as well (presumably `false`).